gap/base/recognition.gi

```diff
@@ -659,6 +660,15 @@ InstallGlobalFunction( RecogniseGeneric,
     return ri;
   end);
 
+InstallGlobalFunction( ValidateHomomInput,
+  function(ri,x)
+    if Hasvalidatehomominput(ri) then
+        return validatehomominput(ri)(x);
```

Comment:
I wonder if we shouldn't pass `ri` to the validation function, too, for more flexibility in the future. I mean, sure, if you create that validation function as a lambda on the spot each time, you don't need that. But may be more convenient (e.g. for debugging, and for more complex validation) to define a single such function in a central place, and then pass `ri` to it.
```suggestion
        return validatehomominput(ri)(ri, x);
```